test/functional/services/common/browser.ts

```diff
@@ -177,6 +182,28 @@ class BrowserService extends FtrService {
     return currentWithoutTime;
   }
 
+  /**
+   * Uses the 'retry' service and waits for the current browser URL to match the provided path.
+   * NB the provided path can contain query params as well as hash anchors.
+   * Using retry logic makes URL assertions less flaky
+   * @param expectedPath The relative path that we are expecting the browser to be on
+   * @returns a Promise that will reject if the browser URL does not match the expected one
+   */
+  public async waitForUrlToBe(expectedPath: string) {
+    const retry = await this.ctx.getService('retry');
+    const log = this.ctx.getService('log');
+
+    return retry.waitFor(`URL to be ${expectedPath}`, async () => {
+      const currentUrl = await this.getCurrentUrl();
+      const { path, search, hash } = Url.parse(currentUrl);
+      const currentPath = `${path}${search ?? ''}${hash ?? ''}`;
+
+      if (currentPath !== expectedPath)
+        log.debug(`Expected URL to be ${expectedPath}, got ${currentPath}`);
```

Comment:
Just a thought: is it possible that for a short period of time `currentPath` will match the `expectedPath` but due to redirect the other second user will be on the different page.
Tbh I don't know if should handle it here or have a different method to verify url consistency.

Reply:
In this case, I believe it would make sense to specify the "after redirect" URL as `expectedPath`. This way, the check will eventually succeed, thanks to the `retry.waitFor`.

Now, if somebody introduces a redirect after the test has been written, I agree it could cause a flaky behaviour.

We could add some initial delay to the check, or we could also confirm that the URL stays the same for some _timeMillis_, but that would not ensure that there isn't some logic on the target page that will redirect after some time, so I don't think we can guarantee it 100%.

I can try to make it more robust by adding a double check + delay between checks:
* Check URL matches `expectedPath`
* await delay(millis)
* Check again URL matches `expectedPath`

WDYT?